modules/gdscript/gdscript_parser.cpp

```diff
@@ -3482,11 +3482,13 @@ void GDScriptParser::_parse_class(ClassNode *p_class) {
 						Variant constant = tokenizer->get_token_constant();
 						String icon_path = constant.operator String();
 
+#ifdef TOOLS_ENABLED
```

Comment:
Does any of the surrounding logic make sense in an exported game? IMO the GDScript parser should just discard all this editor-specific information in the runtime.

On top of checking `TOOLS_ENABLED`, this code should likely also check `Engine::get_singleton()->is_editor_hint()`, as the code is also not relevant when running the game with a `TOOLS_ENABLED` binary but not in the editor (or is it, for things like the remote debugger?).

Reply:
The parser still needs to consume the tokens, but I agree that it doesn't need to fetch nor store the information.

Reply:
I have now updated it to wrap everything but the if statements and `advance` operations in `TOOLS_ENABLED` and an `Engine::get_singleton()->is_editor_hint()` check.